src/rebar_hex_repos.erl

```diff
@@ -86,13 +89,18 @@ merge_repos(Repos) ->
                 end, [], Repos).
 
 update_organizations(Repos) ->
-    lists:map(fun(Repo=#{organization := Organization,
+    lists:map(fun(Repo=#{organization := _Organization,
                          parent := ParentName}) ->
                       {ok, Parent} = get_repo_config(ParentName, Repos),
-                      ParentRepoUrl = rebar_utils:to_list(maps:get(repo_url, Parent)),
```

Comment:
What url is it getting here?This should result in the default repos url in the case that parent is `hexpm`. Makes me think the fix is figuring out why it isn't getting default repos url here.

Reply:
Per the issue:
 `rebar_hex_repos` currently tries to build up a URL for non-default repos, then in turn `hex_core`
      does the same, thus we end up with `https://repo.hex.pm/repos/foo?/repos/foo/packages/bar`.
      We simply need to set the repo_url for non default repos to the user defined
      repo url or fall back to https://repo.hex.pm if the user defined repo url is not set.

Does that demystify the change?